adaptive_navigation/lib/scaffold.dart

```diff
@@ -52,105 +46,132 @@ class AdaptiveNavigationScaffold extends StatefulWidget {
+      return NavigationType.navigationRail;
+    } else {
+      return NavigationType.bottomNavigation;
+    }
+  }
+
   @override
   Widget build(BuildContext context) {
-    // Show a Drawer
-    if (_isLargeScreen(context)) {
-      return Row(
-        children: [
-          Drawer(
-            child: Column(
-              children: [
-                DrawerHeader(
-                  child: Center(
-                    child: widget.title,
-                  ),
+    final NavigationTypeResolver navigationTypeResolver = widget.navigationTypeResolver ?? _defaultNavigationTypeResolver;
+    switch(navigationTypeResolver(context)) {
+      case NavigationType.bottomNavigation:
+        // Show a Scaffold with a BottomNavigationBar.
+        return Scaffold(
+          body: widget.body,
+          appBar: AppBar(title: widget.title),
+          bottomNavigationBar: BottomNavigationBar(
+            items: [
+              ...widget.destinations.map(
+                    (d) => BottomNavigationBarItem(
+                  icon: Icon(d.icon),
+                  title: Text(d.title),
                 ),
-                for (var d in widget.destinations)
-                  ListTile(
-                    leading: Icon(d.icon),
-                    title: Text(d.title),
-                    selected:
-                        widget.destinations.indexOf(d) == widget.currentIndex,
-                    onTap: () => _destinationTapped(d),
-                  ),
-              ],
-            ),
+              ),
+            ],
+            currentIndex: widget.currentIndex,
+            onTap: widget.onNavigationIndexChange,
           ),
-          VerticalDivider(
-            width: 1,
-            thickness: 1,
-            color: Colors.grey[300],
+          floatingActionButton: widget.floatingActionButton,
+        );
+      case NavigationType.navigationRail:
+        // Show a Scaffold with a body containing a NavigationRail.
+        return Scaffold(
+          appBar: AppBar(
+            title: widget.title,
           ),
-          Expanded(
-            child: Scaffold(
-              appBar: AppBar(),
-              body: widget.body,
-              floatingActionButton: widget.floatingActionButton,
-            ),
+          body: Row(
+            children: [
+              NavigationRail(
+                leading: widget.floatingActionButton,
+                destinations: [
+                  ...widget.destinations.map(
+                        (d) => NavigationRailDestination(
+                      icon: Icon(d.icon),
+                      label: Text(d.title),
+                    ),
+                  ),
+                ],
+                selectedIndex: widget.currentIndex,
+                onDestinationSelected: widget.onNavigationIndexChange ?? (_) {},
+              ),
+              VerticalDivider(
+                width: 1,
+                thickness: 1,
+                color: Colors.grey[300],
```

Comment:
I think hard coding the divider color may not be a good idea

Reply:
removed